be/src/io/cache/file_block.cpp

```diff
@@ -161,32 +161,27 @@ Status FileBlock::read(Slice buffer, size_t read_offset) {
     return _mgr->_storage->read(_key, read_offset, buffer);
 }
 
-Status FileBlock::change_cache_type_by_mgr(FileCacheType new_type) {
+Status FileBlock::change_cache_type_between_ttl_and_others(FileCacheType new_type) {
     std::lock_guard block_lock(_mutex);
     DCHECK(new_type != _key.meta.type);
-    if (_download_state == State::DOWNLOADED) {
-        KeyMeta new_meta;
-        new_meta.expiration_time = _key.meta.expiration_time;
-        new_meta.type = new_type;
-        auto st = _mgr->_storage->change_key_meta(_key, new_meta);
-        TEST_SYNC_POINT_CALLBACK("FileBlock::change_cache_type", &st);
-        if (!st.ok()) return st;
-    }
+    DCHECK(new_type == FileCacheType::TTL || _key.meta.type == FileCacheType::TTL);
```

Comment:
can we return error instead of DCHECK?

Reply:
It is an assertion (warning to future developers) instead of error handling (warning to users), so I think no we cannot.

Reply:
On a second thought, I should add a warning for provide more info in release version.